core/src/main/java/jenkins/util/Listeners.java

```diff
@@ -42,20 +43,42 @@
      * <p>Only suitable for listener methods which do not throw checked or otherwise documented exceptions.
      * @param <L> the type of listener
      * @param listenerType the type of listener
+     * @param asSystem whether to impersonate {@link ACL#SYSTEM2}.
+     *                 For most listener methods, this should be {@code true},
+     *                 so that listener implementations can freely perform various operations without access checks.
+     *                 In some cases, existing listener interfaces were implicitly assumed to pass along user authentication,
+     *                 because they were sometimes triggered by user actions such as configuration changes;
+     *                 this is an antipattern (better to pass an explicit {@link Authentication} argument if relevant).
      * @param notification a listener method, perhaps with arguments
+     * @since TODO
      */
-    public static <L> void notify(Class<L> listenerType, Consumer<L> notification) {
-        try (ACLContext ctx = ACL.as2(ACL.SYSTEM2)) {
+    public static <L> void notify(Class<L> listenerType, boolean asSystem, Consumer<L> notification) {
+        Runnable r = () -> {
             for (L listener : ExtensionList.lookup(listenerType)) {
                 try {
                     notification.accept(listener);
                 } catch (Throwable x) {
                     Logger.getLogger(listenerType.getName()).log(Level.WARNING, null, x);
                 }
             }
+        };
+        if (asSystem) {
+            try (ACLContext ctx = ACL.as2(ACL.SYSTEM2)) {
+                r.run();
+            }
+        } else {
+            r.run();
         }
     }
 
+    /**
+     * @deprecated call {@link #notify(Class, boolean, Consumer)}
+     */
+    @Deprecated
+    public static <L> void notify(Class<L> listenerType, Consumer<L> notification) {
```

Comment:
probably not used anywhere in the wild? it came in a very recent weekly can we just bin it?

Reply:
Probably, just wanted to err on the safe side.